ProjFS.Mac/PrjFSKext/PrjFSKext/PerformanceTracing.cpp

```diff
@@ -33,31 +30,38 @@ IOReturn PerfTracing_ExportDataUserClient(IOExternalMethodArguments* arguments)
 #endif
 }
 
-void PerfTracing_RecordSample(PerfTracingProbe* probe, uint64_t startTime, uint64_t endTime)
+void PerfTracing_RecordSample(PrjFSPerfCounter counter, uint64_t startTime, uint64_t endTime)
 {
-    uint64_t interval = endTime - startTime;
+    PerfTracingProbe* probe = &profile_probes[counter];
     
-    atomic_fetch_add(&probe->numSamples1, 1);
-    atomic_fetch_add(&probe->sum, interval);
+    uint64_t interval = endTime - startTime;
     
-    __uint128_t intervalSquared = interval;
-    intervalSquared *= intervalSquared;
-    atomic_fetch_add(&probe->sumSquares, intervalSquared);
+    atomic_fetch_add(&probe->numSamples, 1);
     
-    // Update minimum sample if necessary
+    if (0 != interval)
     {
-        uint64_t oldMin = atomic_load(&probe->min);
-        while (interval < oldMin && !atomic_compare_exchange_weak(&probe->min, &oldMin, interval))
-        {}
-    }
+        atomic_fetch_add(&probe->sum, interval);
 
-    // Update maximum sample if necessary
-    {
-        uint64_t oldMax = atomic_load(&probe->max);
-        while (interval > oldMax && !atomic_compare_exchange_weak(&probe->max, &oldMax, interval))
-        {}
-    }
+        __uint128_t intervalSquared = interval * interval;
```

Comment:
Thanks for pushing the fix for this. I was just "cleaning up" the original code because the first line said `intervalSquared = interval`, which struck me as false, and this side effect of refactoring the code was not super obvious. Your latest push makes it much more clear.